static/js/components/dashboard/CourseListCard.js

```diff
@@ -11,54 +11,145 @@ import type {
 } from '../../flow/couponTypes';
 import CourseRow from './CourseRow';
 import FinancialAidCalculator from '../../containers/FinancialAidCalculator';
+import type { CoursePrice } from '../../flow/dashboardTypes';
+import type { CourseRun } from '../../flow/programTypes';
+import {
+  FA_TERMINAL_STATUSES,
+  COUPON_CONTENT_TYPE_PROGRAM,
+  COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT,
+  COUPON_AMOUNT_TYPE_FIXED_DISCOUNT,
+} from '../../constants';
+import {
+  makeAmountMessage,
+  makeCouponReason,
+} from '../../lib/coupon';
+import { pickExistingProps } from '../../util/util';
 
 export default class CourseListCard extends React.Component {
   props: {
-    coupon:                       ?Coupon,
+    coupon?:                      Coupon,
     program:                      Program,
-    courseEnrollAddStatus?:       string,
+    coursePrice:                  CoursePrice,
     prices:                       CalculatedPrices,
     openFinancialAidCalculator?:  () => void,
     now?:                         Object,
-    addCourseEnrollment:          (courseId: string) => void,
-    openCourseContactDialog:      (course: Course, canContactCourseTeam: boolean) => void
+    addCourseEnrollment:          (courseId: string) => Promise<*>,
+    openCourseContactDialog:      (course: Course, canContactCourseTeam: boolean) => void,
+    setEnrollSelectedCourseRun:   (r: CourseRun) => void,
+    setEnrollCourseDialogVisibility: (bool: boolean) => void,
   };
 
+  renderPriceMessage() {
+    const {
+      program,
+      coupon,
+      prices,
+    } = this.props;
+
+    // Special case: 100% off coupon
+    if (coupon &&
+        coupon.content_type === COUPON_CONTENT_TYPE_PROGRAM &&
+        coupon.amount_type === COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT &&
+        coupon.amount.equals(1)
+    ) {
+      return <p className="price-message">
+        Courses in this program are free, because of your coupon.
+      </p>;
+    }
+
+    if (program.financial_aid_availability) {
+      const financialAid = program.financial_aid_user_info;
+      if (FA_TERMINAL_STATUSES.includes(financialAid.application_status)) {
+        // all prices in the map are the same, so just get a random one
+        const iter = prices.values();
+        const { value: price } = iter.next();
+        if (!price) {
+          return null;
+        }
+        if (coupon && coupon.content_type === COUPON_CONTENT_TYPE_PROGRAM) {
+          // financial aid + coupon
+          return <p className="price-message">
+            Your price is <strong>${ price.toString() } USD per course,</strong> including
+            both financial aid and your coupon.
+            If you want to audit courses for FREE and upgrade later,
+            click Enroll Now then choose the audit option.
+          </p>;
+        } else {
+          return <p className="price-message">
+            Your Personal Course Price is <strong>${ price.toString() } USD per course.</strong>
```

Comment:
Oops, JSX isn't as smart as I'd like it to be about whitespace. The problem isn't the line ending with a period, it's the line ending with a starting or ending HTML tag. I think I've fixed it, but it might be worth double-checking.